effects/cargo/default.nix

```
@@ -0,0 +1,36 @@
+{ lib
+, mkEffect
+, cargo
+, cargoSetupHook
+, rustc
+, clang
+, llvmPackages
+}:
+args@{ secretName ? throw ''effects.cargo: You must provide `secretName`, the name of the secret which holds the "${secretField}" field.''
+, secretField ? "token"
+, secretsMap ? { }
+, manifestPath ? null
```

I think this must be required, but we can make it a bit easier.

```suggestion
, manifestPath ? "${src}/Cargo.toml"
, src
```

Now `src` is required, and we automatically avoid the mistake of passing only `./Cargo.toml` or `"${./Cargo.toml}"` which only provides that single file.